superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -338,23 +343,27 @@ export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
                   disabled={newOrOverwrite !== 1}
                 />
               </Radio>
-              <Radio className="sdm-radio" value={2}>
-                {t('Overwrite existing')}
-                <AutoComplete
-                  className="sdm-autocomplete"
-                  options={userDatasetOptions}
-                  onSelect={handleOverwriteDatasetOption}
-                  onSearch={handleSaveDatasetModalSearch}
-                  onChange={value => {
-                    setDatasetToOverwrite({});
-                    setAutocompleteValue(value);
-                  }}
-                  placeholder={t('Select or type dataset name')}
-                  filterOption={filterAutocompleteOption}
-                  disabled={newOrOverwrite !== 2}
-                  value={autocompleteValue}
-                />
-              </Radio>
+              <div className="sdm-overwrite-container">
+                <Radio className="sdm-radio" value={2}>
+                  {t('Overwrite existing')}
+                </Radio>
+                <div className="sdm-autocomplete">
+                  <Select
+                    allowClear
+                    showSearch
+                    placeholder={t('Select or type dataset name')}
+                    ariaLabel={t('Existing dataset')}
+                    onChange={handleOverwriteDatasetOption}
+                    options={input => loadDatasetOverwriteOptions(input)}
+                    value={selectedDatasetToOverwrite}
+                    filterOption={filterAutocompleteOption}
+                    disabled={newOrOverwrite !== 2}
+                    // @ts-ignore
+                    getPopupContainer={() => document.body}
```

Comment:
`document.body` is the default value for `getPopupContainer` so you don't need to set it here 😉 

<img width="1240" alt="Screen Shot 2022-06-27 at 6 01 45 PM" src="https://user-images.githubusercontent.com/70410625/176034952-cfbd90eb-30a1-4098-a5ff-c622597d2960.png">

Reply:
Oh, forget it. I just saw that you're overriding the default behavior of the component 😉 You can leave as it is 👍🏼 I need some coffee

Reply:
If I remove it then `Select` component will use its current behavior (parent node): https://github.com/apache/superset/blob/master/superset-frontend/src/components/Select/Select.tsx#L704